educator/src/Dscp/Educator/Web/Student/Error.hs

```diff
@@ -104,15 +98,22 @@ instance ToJSON APIError where
 -- | Get HTTP error code of error.
 toServantErrNoReason :: APIError -> ServantErr
 toServantErrNoReason = \case
-    SubmissionMalformed{}      -> err403
     BadSubmissionSignature{}   -> err403
     DeletingGradedSubmission{} -> err403
-    EntityAbsent{}             -> err404
+    EntityAbsent err           -> case err of
+        CourseDoesNotExist{}                  -> err404
+        StudentDoesNotExist{}                 -> err404
+        AssignmentDoesNotExist{}              -> err404
+        StudentWasNotEnrolledOnTheCourse{}    -> err404
+        StudentWasNotSubscribedOnAssignment{} -> err404
+        SubmissionDoesNotExist{}              -> err404
+        TransactionDoesNotExist{}             -> err404
+        BlockWithIndexDoesNotExist{}          -> err500
```

Comment:
Why 500?

Reply:
It seems to be unchecked error despite why it's thrown, client should not know about it